src/test/java/com/revature/services/ReviewServiceTest.java

```diff
@@ -96,6 +120,23 @@ void testAdd_Failure_ProductNotFound() {
 		}
 	}
 
+	@Test
+	void testAdd_Failure_DuplicateReview() {
+		ReviewRequest request = new ReviewRequest(this.dummyProduct.getId(), 1, "Not working",
+				"It doesn't work as advertised. I am dissatisfied with this product.");
+		int productId = this.dummyProduct.getId();
+		given(this.pServ.findById(productId)).willReturn(Optional.of(this.dummyProduct));
+		given(this.rServ.findByProductId(this.dummyProduct.getId())).willReturn(Collections.singletonList(
+				new Review(2, 1, "A title", "A review of a thing.", null, null, this.dummyProduct, this.dummyUser)));
+		
+		try {
+			this.rServ.add(request, dummyUser);
+			fail("Expected an exception to be thrown");
```

Comment:
You should specify which exception type is expected to be thrown. In this case, it's `DuplicateReviewException`